javav2/usecases/creating_fsa_app/src/main/java/com/example/fsa/services/DetectSentimentService.java

```diff
@@ -0,0 +1,68 @@
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+package com.example.fsa.services;
+import org.json.simple.JSONObject;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.comprehend.ComprehendClient;
+import software.amazon.awssdk.services.comprehend.model.ComprehendException;
+import software.amazon.awssdk.services.comprehend.model.DetectDominantLanguageRequest;
+import software.amazon.awssdk.services.comprehend.model.DetectDominantLanguageResponse;
+import software.amazon.awssdk.services.comprehend.model.DetectSentimentRequest;
+import software.amazon.awssdk.services.comprehend.model.DetectSentimentResponse;
+import software.amazon.awssdk.services.comprehend.model.DominantLanguage;
+import java.util.List;
+
+public class DetectSentimentService {
+
+    private ComprehendClient getClient() {
+        return ComprehendClient.builder()
+            .region(Region.US_EAST_1)
+            .build();
+    }
+
+    public JSONObject detectSentiments(String text){
+        try {
+            String languageCode = detectTheDominantLanguage(text);
+            DetectSentimentRequest detectSentimentRequest = DetectSentimentRequest.builder()
+                .text(text)
+                .languageCode(languageCode)
+                .build();
+
+            DetectSentimentResponse detectSentimentResult = getClient().detectSentiment(detectSentimentRequest);
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put("sentiment", detectSentimentResult.sentimentAsString());
+            jsonObject.put("language_code", languageCode);
+            return jsonObject;
+
+        } catch (ComprehendException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
```

Comment:
If the program exits on failure, why return null on line 44?

Reply:
Also, is it safe to hard-exit in a lambda handler? Shouldn't this re-throw to the lambda runtime to handle?

Reply:
Fixed this catch to log exception